pkg/api/v1/router_server_components.go

```diff
@@ -2,18 +2,71 @@ package serverservice
 
 import (
 	"github.com/gin-gonic/gin"
+	"github.com/volatiletech/null/v8"
+	"github.com/volatiletech/sqlboiler/v4/boil"
 	"github.com/volatiletech/sqlboiler/v4/queries/qm"
 )
 
+// serverComponentList returns a response with the list of components that matched the params.
 func (r *Router) serverComponentList(c *gin.Context) {
+	pager := parsePagination(c)
```

Comment:
similar to what @diogomatsubara did in https://github.com/metal-toolbox/hollow-serverservice/pull/48/files , seems to me like we would want pagination to already happen [on the SQL query](https://github.com/metal-toolbox/hollow-serverservice/pull/48/files) , that way, we would reduce the memory footprint from this server and be more resilient to DDoS's from this side.

Reply:
@JAORMX ack, let me see how that can be fixed. Which SQL query are you referring to in your comment, the link leads me to #48

Reply:
@joelrebel copypasta error, here it is: https://github.com/metal-toolbox/hollow-serverservice/blob/6297dc0251f020f85a9576c700ad72bf53b683bc/pkg/api/v1/router_firmware.go#L31

Reply:
@JAORMX we currently have a [default limit](https://github.com/metal-toolbox/hollow-serverservice/blob/ec5311e83f5978442769aed93c7f5a0cab8a1daa/pkg/api/v1/pagination.go#L14) of 100 items, it does seem to be on the higher side though. Are you suggesting pagination be done within the default limit?


```
SELECT "server_components".* FROM "server_components" LEFT JOIN versioned_attributes as 
... <snip> ...
AND json_extract_path_text(server_components_ver_attr_0.data::JSONB, $4) = $5)) ORDER BY server_components.created_at DESC LIMIT 100;    
```